autogluon/utils/tabular/ml/trainer/abstract_trainer.py

```diff
@@ -520,25 +533,27 @@ def train_multi_levels(self, X_train, y_train, X_val, y_val, models: List[Abstra
+                base_model_names, _ = self.stack_new_level(X=X_train, y=y_train, X_val=X_val, y_val=y_val, models=models, level=level, base_model_names=base_model_names, hyperparameter_tune=hyperparameter_tune, feature_prune=feature_prune, time_limit_core=time_limit_core, time_limit_aux=time_limit_aux)
             else:
-                self.stack_new_level(X=X_train, y=y_train, X_val=X_val, y_val=y_val, level=level, time_limit_core=time_limit_core, time_limit_aux=time_limit_aux)
+                models = self.get_models(hyperparameters, level=level)
+                base_model_names, _ = self.stack_new_level(X=X_train, y=y_train, X_val=X_val, y_val=y_val, models=models, level=level, base_model_names=base_model_names, time_limit_core=time_limit_core, time_limit_aux=time_limit_aux)
 
         self._time_limit = None
         self.save()
 
-    def stack_new_level(self, X, y, X_val=None, y_val=None, level=0, models=None, hyperparameter_tune=False, feature_prune=False, time_limit_core=None, time_limit_aux=None):
-        core_models = self.stack_new_level_core(X=X, y=y, X_val=X_val, y_val=y_val, models=models, level=level, hyperparameter_tune=hyperparameter_tune, feature_prune=feature_prune, time_limit=time_limit_core)
+    def stack_new_level(self, X, y, models: List[AbstractModel], X_val=None, y_val=None, level=0, base_model_names=None, hyperparameter_tune=False, feature_prune=False, time_limit_core=None, time_limit_aux=None):
+        core_models = self.stack_new_level_core(X=X, y=y, X_val=X_val, y_val=y_val, models=models, level=level, base_model_names=base_model_names, hyperparameter_tune=hyperparameter_tune, feature_prune=feature_prune, time_limit=time_limit_core)
         if self.bagged_mode:
-            aux_models = self.stack_new_level_aux(X=X, y=y, level=level+1, time_limit=time_limit_aux)
+            aux_models = self.stack_new_level_aux(X=X, y=y, level=level+1, base_model_names=core_models, time_limit=time_limit_aux)
         else:
-            aux_models = self.stack_new_level_aux(X=X_val, y=y_val, fit=False, level=level + 1, time_limit=time_limit_aux)
-        return core_models + aux_models
+            aux_models = self.stack_new_level_aux(X=X_val, y=y_val, fit=False, level=level + 1, base_model_names=core_models, time_limit=time_limit_aux)
+        return core_models, aux_models
 
-    def stack_new_level_core(self, X, y, X_val=None, y_val=None, models=None, level=1, stack_name='core', kfolds=None, n_repeats=None, hyperparameter_tune=False, feature_prune=False, time_limit=None, save_bagged_folds=None, stacker_type=StackerEnsembleModel, extra_ag_args_fit=None):
+    def stack_new_level_core(self, X, y, models: List[AbstractModel], X_val=None, y_val=None, level=1, base_model_names=None, stack_name='core', kfolds=None, n_repeats=None, hyperparameter_tune=False, feature_prune=False, time_limit=None, save_bagged_folds=None, stacker_type=StackerEnsembleModel):
```

Comment:
documentation should clarify what core model is

Reply:
added